src/ApiManagement/ApiManagement.ServiceManagement/Commands/removeApiManagementApiDiagnostic.cs

```diff
@@ -0,0 +1,66 @@
+﻿//  
+// Copyright (c) Microsoft.  All rights reserved.
+// 
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+// 
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+namespace Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Commands
+{
+    using Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Models;
+    using System;
+    using System.Management.Automation;
+
+    [Cmdlet("Remove", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ApiManagementApiDiagnostic")]
```

Comment:
Let's use words `Attach` and `Detach` instead of `Set` and `Remove`, as these fits more to the actual outcome of these commands.

Reply:
also, it's not `ApiManagementApiDiagnostic` but `ApiManagementApiDiagnosticLogger`

Reply:
I totally agree about the  `Attach` one. Instead of Detach I suggest to use `Disable`. Actually we don't detach but we disable the API Diagnostic from the API.